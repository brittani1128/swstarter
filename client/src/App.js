import React from "react";
// import logo from './logo.svg';
import Main from "./pages/Main/Main.js";
import "./App.css";

class App extends React.Component {
  state = {
    data: null
  };
  componentDidMount() {
    // Call our fetch function below once the component mounts
    this.callBackendAPI()
      .then(res => this.setState({ data: res.express }))
      .catch(err => console.log(err));
  }
  // Fetches our GET route from the Express server. (Note the route we are fetching matches the GET route from server.js
  callBackendAPI = async () => {
    const response = await fetch("/express_backend");
    const body = await response.json();

    if (response.status !== 200) {
      throw Error(body.message);
    }
    return body;
  };

  render() {
    return (
      <div className="App">
        <header className="header" />
        <div className="container">
          <Main />
        </div>
      </div>
    );
  }
}

export default App;
