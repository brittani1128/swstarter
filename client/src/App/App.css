.header {
  height: 50px;
  box-shadow: 0 2px 0 0 var(--white);
  background-color: #ffffff;
  display: flex;
  align-content: center;
}

.container {
  width: 900px;
  margin: auto;
  margin-top: 30px;
}

.header h2 {
  font-family: Montserrat;
  font-size: 18px;
  font-weight: bold;
  font-style: normal;
  color: #0ab463;
  margin: 0px auto;
  padding-top: 14px;
  text-align: center;
}
