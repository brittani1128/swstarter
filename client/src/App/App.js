import React from "react";
import { Route, Switch } from "react-router-dom";
import Main from "../pages/Main/Main.js";
import Details from "../pages/Details/Details.js";
import "./App.css";

class App extends React.Component {
  state = {
    data: null
  };
  // componentDidMount() {
  //   // Call our fetch function below once the component mounts
  //   this.callBackendAPI()
  //     .then(res => this.setState({ data: res.express }))
  //     .catch(err => console.log(err));
  // }
  // // Fetches our GET route from the Express server. (Note the route we are fetching matches the GET route from server.js
  // callBackendAPI = async () => {
  //   const response = await fetch("/express_backend");
  //   const body = await response.json();

  //   if (response.status !== 200) {
  //     throw Error(body.message);
  //   }
  //   return body;
  // };

  render() {
    return (
      <div className="App">
        <header className="header">
          <h2>SWStarter</h2>
        </header>
        <div className="container">
          <Switch>
            <Route exact path="/" component={Main} />
            <Route path="/details" component={Details} />
          </Switch>
        </div>
      </div>
    );
  }
}

export default App;
