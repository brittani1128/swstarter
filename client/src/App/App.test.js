import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import App from "./App";

// test instance methods
// shallow.instance
// look up way to mock out context
//

describe("App component", () => {
  it("should render", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleSelect", () => {
    const mockFunction = jest.fn();
    it("should set the state of searchQuery to event value", () => {});
    // test state change
    const instance = shallow(<App handleSubmit={mockFunction} />).instance();
    instance.setState = sinon.spy();
    instance.handleSelect();
    expect(instance.setState.calledWith({}));
  });

  describe("handleSubmit", () => {
    // test selectquery function called
    const mockFunction = jest.fn();
    it("should call selectQuery when clicked", () => {
      const spy = sinon.spy();
    });
  });

  //   describe("search input", () => {
  //     it("should respond to change event and change state of searchQuery", () => {
  //       const wrapper = shallow(<App />);
  //       wrapper
  //         .find("#input")
  //         .simulate("change", { target: { name: "input", value: "yoda" } });
  //       expect(wrapper.state("resultSearchQuery")).toEqual("yoda");
  //     });
  //   });
});
