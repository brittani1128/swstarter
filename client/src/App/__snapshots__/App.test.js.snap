// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h2>
      Star Wars Search
    </h2>
  </header>
  <div
    className="container"
  >
    <ContextProvider
      value={
        Object {
          "detailsSelection": "",
          "handleDetails": [Function],
          "handleInputChange": [Function],
          "handleSelect": [Function],
          "handleSubmit": [Function],
          "resultIndex": null,
          "resultSearchQuery": "",
          "searchQuery": "people",
          "searchResults": Array [],
          "selectQuery": [Function],
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/details"
        />
      </Switch>
    </ContextProvider>
  </div>
</div>
`;

exports[`App component should render App component 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h2>
      Star Wars Search
    </h2>
  </header>
  <div
    className="container"
  >
    <ContextProvider
      value={
        Object {
          "detailsSelection": "",
          "handleDetails": [Function],
          "handleInputChange": [Function],
          "handleSelect": [Function],
          "handleSubmit": [Function],
          "resultIndex": null,
          "resultSearchQuery": "",
          "searchQuery": "people",
          "searchResults": Array [],
          "selectQuery": [Function],
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/details"
        />
      </Switch>
    </ContextProvider>
  </div>
</div>
`;
